### ISSUE TITLE
Check Transpiled JavaScript workflow fails for Dependabot PRs

### ISSUE THREAD

**Issue by arianvp** [actions/typescript-action#986]
Closely related to #957 but different.

When you use this repo as a template, all dependabot PRs that are opened will fail as they do not regenerate the `dist` folder.

Is there a way to make dependabot do that or to not have the weird requirement to commit compiled code? 

Because currently dependabot creates a lot of churn because all the PRs it opens will fail

**Comment by arianvp** [comment]
It would be great if there would be a way to not commit the dist folder. Can’t we publish releases to GitHub Packages or something?

**Comment by ncalteen** [comment]
Yeah it can be a nuisance with dependency updates, since the `dist/` directory can be modified when non-development dependencies are updated. I tested this out a few different ways, and it is possible to set up a GitHub Action that ensures the `dist/` directory is rebuilt on Dependabot PRs, however I opted not to include this in the template because its way too easy to end up with infinite loops when GitHub Actions runs and updates the PR branch.

I'm waiting on [Immutable Actions](https://github.com/github/roadmap/issues/592) to go GA before trying another approach. In the meantime, you may want to checkout the [container action template](https://github.com/actions/container-toolkit-action) instead. This one is pretty straightforward to set up so the container pulls the dependencies and builds as part of your CI pipeline.

**Comment by arianvp** [comment]
That repo has the same issue no? It also commits `dist`. It will suffer from the same problems.

**Comment by ncalteen** [comment]
Yes that is true, sorry. Unfortunately, I don't have a great option for this until immutable actions are live. At that point I should be able to make this easier :) I will keep this issue open for the time being and circle back once that feature is GA.

**Comment by ncalteen** [comment]
Yeah this approach does work well! My main concern with adding it to the template repos is that there is a potential for infinite loops of GitHub Actions workflow runs :(

**Comment by arianvp** [comment]
Hello it seems that Immutable Actions is completely dropped from the roadmap in favour of Immutable Releases

https://github.blog/news-insights/product-news/whats-coming-to-our-github-actions-2026-security-roadmap/

This means this issue will not be addressed it seems.

Any update on this?

How am I supposed to keep a GitHub Action up to date if it doesn't work with Dependabot?
